packages/babel-core/src/config/full.ts

```diff
@@ -206,6 +221,7 @@ export default gensync<(inputOpts: unknown) => ResolvedConfig | null>(
     return {
       options: opts,
       passes: passes,
+      externalDependencies: new ReadonlySet(externalDependencies),
```

Comment:
I think it suffices to use TS `ReadOnlyArray` for `externalDependencies`, since eventually they would be added to a Set.